docs/core/tutorials/nft-tutorials/whitelists.md

```diff
@@ -0,0 +1,316 @@
+---
+title: Merkle Tree Whitelist
+displayed_sidebar: coreSidebar
+---
+
+# Merkle Tree NFT Whitelist on CoreSpace using Hardhat
+
+Using a Merkle Tree whitelist provides a cost-efficient way to distribute NFTs to a predefined list of addresses. It ensures that only whitelisted addresses can mint the NFTs, enhancing security and efficiency. Storing a whitelist with thousands of addresses in a smart contract would incur high gas fees for each update. However, since in the Merkle Tree verification process, leaf nodes and proofs can be stored off-chain while only the root value needs to be stored on-chain, this method significantly saves on gas costs.
```

Comment:
```suggestion
Using a Merkle Tree whitelist provides a cost-efficient way to distribute NFTs to a predefined list of addresses. It ensures that only whitelisted addresses can mint the NFTs, enhancing security and efficiency. Writing a whitelist of thousands of addresses directly within a smart contract would result in significant gas costs, and additionally, due to the maximum transaction gas limit restriction, it might be impractical to update such a large whitelist in a single transaction. However, since in the Merkle Tree verification process, leaf nodes and proofs can be stored off-chain while only the root value needs to be stored on-chain, this method significantly saves on gas costs and enables whitelist setup with only one transaction.
```